### DISCUSSION TITLE
Neon with quantum-corrected PR

### DISCUSSION THREAD

**Discussion by ianhbell** [ClapeyronThermo/Clapeyron.jl#209] [Q&A] [answered]
In the paper of Aasen they select the FH1 model for neon, but Clapeyron uses the empirical parameters instead. Is there a reason why? Thermopack seems to match the values in the paper

**Comment by pw0908** [comment]
There wasn't a particular reason. I think we were just benchmarking and picked one set of parameters. You can specify your own set of parameters using the `userlocations` argument to use the FH1 set of parameters. At the end of the day, the functional form for beta remains unchanged and the AADs weren't that massively different. We couldve probably supported the explicit FH1 and FH2 forms for this cubic, but these lose the benefits of a cubic.

**Comment by ianhbell** [comment]
I guess my thought was if you are going to pick one it would make sense to follow their model selection to match the defaults in ThermoPack (what I am comparing against as we speak). I added this model to teqp a couple of days ago.

**Comment by pw0908** [comment]
We just benchmarked against the paper at the time since there was no thermopack back then. I am wondering why the empirical model wasn't the best for all cases? Since the FH1 and FH2 are just further-constrained versions of the empirical model.

**Comment by ianhbell** [comment]
I agree I had the same question

**Comment by pw0908** [comment]
This is more of a discussion than an issue; I'll just convert it

**Comment by ianhbell** [comment]
Fair enough - could you provide an example of setting parameters at runtime with ``userlocations``? I am trying to do complete validation

**Reply by pw0908** [reply, marked as answer]
Something like this:
```julia
model = QCPR(["helium"]; userlocations = (;Tc = [5.1953], Pc = [227600], Mw = [4.0]),
                   alpha_userlocations = (;L = [0.18976], M = [1.3964], N = [0.58143]),
             translation_userlocations = (;v_shift = [-2.9291e-6]),
                  mixing_userlocations = (;A = [1.8774], B = [7.7564],l = [0.;;]))
```

**Reply by pw0908** [reply]
Technically we didnt need to specify the critical point data since that's already in the database

**Reply by ianhbell** [reply]
Thanks that is really helpful. I might recommend similar examples for each model type, otherwise it is not intuitive how to instantiate the model with your own parameters

**Reply by pw0908** [reply]
Currently revamping the docs in a separate branch. There are a few different ways one can specify their own parameters. This is just one of the more-direct ones.

**Reply by ianhbell** [reply]
Docs are some of the most time consuming things to develop for a library, but the most useful for others

**Comment by ianhbell** [comment]
How about for a mixture? I can't make out how to init a binary. My best guess was something like this:
``` julia
model = Clapeyron.QCPR(["neon","hydrogen"]; 
    userlocations = (;Tc = [44.492,33.19], Pc = [2679000, 1296400], Mw = [4.0, 4]),
    alpha_userlocations = (;L = [0.40453, 156.21], M = [0.95861, -0.0062072], N = [0.8396, 5.047]),
    translation_userlocations = (;v_shift = [-2.4665e-6, -3.8139e-6]),
    mixing_userlocations = (;A = [0.4673, 3.0696], B = [2.4634, 12.682], l = [0.,0.0;;]),
    k = [[0.,0.18], [0.18,0.0]],
    l = [[0.,0], [0, 0.0]]
)
T = 300.0
z = [0.5, 0.5]
Clapeyron.cross_second_virial(model, T, z)
```
But that doesn't work, with this error:
```
syntax: unexpected semicolon in array expression around In[32]:1

Stacktrace:
 [1] top-level scope
   @ In[32]:1
```

**Reply by pw0908** [reply]
You've placed the binary interaction parameters in the wrong place:
```julia
model = Clapeyron.QCPR(["neon","hydrogen"]; 
                  userlocations = (;Tc = [44.492,33.19], Pc = [2679000, 1296400], Mw = [4.0, 4], k = [0. 0.18; 0.18 0.]),
                  alpha_userlocations = (;L = [0.40453, 156.21], M = [0.95861, -0.0062072], N = [0.8396, 5.047]),
                  translation_userlocations = (;v_shift = [-2.4665e-6, -3.8139e-6]),
                  mixing_userlocations = (;A = [0.4673, 3.0696], B = [2.4634, 12.682], l = [0.0 0.0;0.0 0.0]))
```

**Reply by ianhbell** [reply]
Aha, that makes sense now. Could make a nice snippet for the docs (hint, hint)

**Reply by pw0908** [reply]
Way ahead of you: https://github.com/ClapeyronThermo/Clapeyron.jl/blob/docs-update/docs/src/tutorials/user_defined_parameters.md
